### ISSUE TITLE
[anecdotal/opinion] report: interspersed branch / uncovered lines are much less readable than 4.x era

### ISSUE THREAD

**Issue by asottile** [coveragepy/coveragepy#955]
in coverage 5.x (or at least, that's when I noticed it) the missing branches are now interspersed with missing lines in the `coverage report` terminal display

Here's an example:

```
babi/highlight.py              389     15    116     11    93%   93->94, 94, 101->102, 102, 109->110, 110, 119->120, 120-121, 122->123, 123-124, 159->160, 160, 163->164, 164, 184-185, 199->200, 200, 201->202, 202, 652->653, 653, 679->680, 680
```

prior to coverage 5, the output looked like this:

```
babi/highlight.py              389     15    116     11    93%   94, 102, 110, 120-121, 123-124, 160, 164, 184-185, 200, 202, 653, 680, 93->94, 101->102, 109->110, 119->120, 122->123, 159->160, 163->164, 199->200, 201->202, 652->653, 679->680
```

I prefer the older output because:
- when I am looking to increase coverage, I'm first going to try and increase the line coverage.  This will always necessarily increase the branch coverage as well
- with branches and lines interspersed, it is difficult to see at a glance what the next line to fix is (this is especially difficult when `##-##` looks very much like `##->##`)

A hacky, and far-from-perfect workaround that I'm using right now is `coverage report | sed -r 's/[0-9]+->[0-9]+//g;s/,( ,)+/,/g'` which (mostly) hides the missed branches until I'm ready to work on the missing branch coverage

**Comment by tolomea** [comment]
I think this enhancement I just suggested would greatly improve the situation for you https://github.com/nedbat/coveragepy/issues/1065

**Comment by tolomea** [comment]
Out of curiosity I looked through your output, none of the branch reports there are adding useful information and they could all be omitted leaving ``` 94, 102, 110, 120-121, 123-124, 160, 164, 184-185, 200, 202, 653, 680 ```

**Comment by nedbat** [comment]
This is now released as part of [coverage 5.5](https://pypi.org/project/coverage/5.5).
